.circleci/config.yml

```diff
@@ -0,0 +1,195 @@
+version: 2.1
+orbs: 
+  browser-tools: circleci/browser-tools@1.4.0
+  moneysmart-orb: moneysmartco/circleci-orb@1.0.16
+  aws-cli: circleci/aws-cli@3.1
+  helm: circleci/helm@1.0
+
+defaults: &defaults
+  working_directory: ~/app
+  docker:
+    # the Docker image with Cypress dependencies
+    - image: cypress/base:16.14.0
+      environment:
+        ## this enables colors in the output
+        TERM: xterm
+aliases:
+  - &prod_aws_creds
+    aws-cli/setup:
+      aws-access-key-id: AWS_ACCESS_KEY
+      aws-secret-access-key: AWS_SECRET_ACCESS_KEY
+      aws-region: AWS_REGION
+  - &stg_aws_creds
+    aws-cli/setup:
+      aws-access-key-id: STAGING_AWS_ACCESS_KEY
+      aws-secret-access-key: STAGING_AWS_SECRET_ACCESS_KEY
+      aws-region: AWS_REGION
+default-image: &default-image
+  machine:
+    image: ubuntu-2004:current
+    docker_layer_caching: true
+docker: &docker-image
+  docker:
+    # the Docker image with Cypress dependencies
+    - image: cimg/python:3.11.4
+      environment:
+        ## this enables colors in the output
+        TERM: xterm
+jobs:
+#---------------------------------------------
+# weaviate Staging Jobs
+#---------------------------------------------
+  get_env_var_from_vault_sg_staging_eks:
+    <<: *docker-image
+    steps:
+      - checkout
+      - *prod_aws_creds
+      - moneysmart-orb/get_env_var_from_vault:
+          secret_path: staging/sg/weaviate
+          repo_name: weaviate-helm
+          output_format: helm
+
+  deploy_to_sg_staging_eks:
+    docker:
+      - image: 'cimg/python:3.10'
+    steps:
+      - checkout
+      - *stg_aws_creds
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Update config
+          command: |
+            aws eks update-kubeconfig --name product-listing-staging
+      - run:
+          name: Update values.yml
+          command: |
+            while IFS=': ' read -r key value; do
+            if [ "$key" ]; then
+                value=$(echo "$value" | tr -d '"')
+                export "$key=$value"
+            fi
+            done < <(grep -E '^\s+\w+\s*:' /tmp/workspace/staging-sg-weaviate-env.yaml)
+            
+            sed -i "s|QUERY_LIMIT|$QUERY_LIMIT|g" weaviate/values-override.yml
+            sed -i "s|AUTHENTICATION_API_KEY|$AUTHENTICATION_API_KEY|g" weaviate/values-override.yml
+            sed -i "s|'MODULES_API_KEY'|"$MODULES_API_KEY"|g" weaviate/values-override.yml
+            sed -i "s|SERVICE_TYPE|$SERVICE_TYPE|g" weaviate/values-override.yml
+      - helm/upgrade-helm-chart:
+          chart: weaviate
+          release-name: ms-labs-weaviate
+          namespace: labs-db
+          values-to-override: image.tag=1.23.0,replicas=3,storage.size=50Gi
```

Comment:
can these come from vault ?
atleast the replicas and storage size ?
and image tag from vaule.yml ?

Reply:
Sure we can add the replicas and storage size to vault